lib/brainstem/api_docs/formatters/open_api_specification/version_2/field_definitions/presenter_field_formatter.rb

```diff
@@ -0,0 +1,147 @@
+require 'brainstem/api_docs'
+require 'brainstem/api_docs/formatters/abstract_formatter'
+require 'brainstem/api_docs/formatters/open_api_specification/helper'
+
+module Brainstem
+  module ApiDocs
+    module Formatters
+      module OpenApiSpecification
+        module Version2
+          module FieldDefinitions
+            class PresenterFieldFormatter < AbstractFormatter
+              include Helper
+              
+              def initialize(presenter, field)
+                @presenter = presenter
+                @field = field
+              end
+
+              def format
+                format_field(@field)
+              end
+              alias_method :call, :format
+
+              private
+              
+              attr_reader :presenter
+
+              def has_properties?(field)
+                field.respond_to?(:configuration)
+              end
+
+              def format_field(field)
+                if field.options[:nested_levels]
+                  format_nested_array_field(field)
+                elsif has_properties?(field)
+                  format_nested_field(field)
+                else
+                  format_simple_field(field)
+                end
+              end
+
+              def format_nested_array_field(field)
+                field_properties_data, nested_levels = format_array_items(field)
+
+                format_nested_array_parent(nested_levels, field_properties_data, format_description(field))
+              end
+
+              def format_array_items(field)
+                field_nested_levels = field.options[:nested_levels]
+
+                if has_properties?(field)
+                  [format_nested_field(field, false), field_nested_levels - 1]
+                else
+                  [type_and_format(field.options[:item_type]), field_nested_levels]
+                end
+              end
+
+              def format_nested_array_parent(nested_level, formatted_data, description = nil)
+                if nested_level == 1
+                  {
+                    'type' => 'array',
+                    'description' => description,
+                    'items' => formatted_data
+                  }
+                else
+                  {
+                    'type' => 'array',
+                    'description' => description,
+                    'items' => format_nested_array_parent(nested_level - 1, formatted_data)
+                  }
+                end.with_indifferent_access.reject { |_, v| v.blank? }
```

Comment:
Do we have active support available to us? If so, we can probably get rid of these `reject { |_, v| v.blank? }`s and replace them with `.compact`

Reply:
Unfortunately, `compact` does not remove blank values :(